pinot-spi/src/main/java/org/apache/pinot/spi/plugin/PluginManager.java

```diff
@@ -244,9 +272,64 @@ public void load(String pluginName, File directory) {
         LOGGER.error("Unable to load plugin [{}] jar file [{}]", pluginName, jarFile, e);
       }
     }
-    PluginClassLoader classLoader = createClassLoader(urlList);
+
+    if (useLegacyPluginClassloader()) {
+      PluginClassLoader classLoader = createClassLoader(urlList);
+      _registry.put(new Plugin(pluginName), classLoader);
+    } else {
+      try {
+        ClassRealm pluginRealm = _classWorld.newRealm(
+                pluginName,
+                new URLClassLoader(urlList.toArray(new URL[0]), ClassLoader.getPlatformClassLoader()));
+
+        ClassLoader pinotLoader = _classWorld.getClassRealm("pinot");
+
+        // All exported packages
+        Stream.of("org.apache.pinot.spi.accounting",
```

Comment:
This list is an assumption. Plugins should only make use of classes in these packages, which is any inside the pinot-spi module. Any other pinot-packages will not be accessible.